src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,106 @@
+package com.internet.mixshop.dao.jdbc;
+
+import com.internet.mixshop.dao.ProductDao;
+import com.internet.mixshop.exception.DataProcessingException;
+import com.internet.mixshop.lib.Dao;
+import com.internet.mixshop.model.Product;
+import com.internet.mixshop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        String query = "INSERT INTO product(name, price) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query,
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, product.getName());
+            statement.setDouble(2, product.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            while (resultSet.next()) {
+                product.setId(resultSet.getLong(1));
+            }
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Product " + product.getName() + " is not added!", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> getById(Long id) {
+        String query = "SELECT * FROM product WHERE product_id = ? AND isDeleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Product product = getProduct(resultSet);
+                return Optional.ofNullable(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Product with id = " + id + " not found", e);
```

Comment:
exception doesn't mean, that there is no product with given id in db
```suggestion
            throw new DataProcessingException("Can't get product with id = " + id, e);
```

Reply:
done